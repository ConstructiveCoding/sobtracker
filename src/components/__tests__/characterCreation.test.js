import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CharacterCreation from '../characterCreation';

describe('CharacterCreation', () => {
  it('should render the character creation screen with no props', () => {
    const styles = {};
    const tree = renderer.create(<CharacterCreation style={styles} />);

    expect(tree).toMatchSnapshot();
  });

  it('should render the character creation screen with props', () => {
    const styles = {};
    const tree = renderer.create(
      <CharacterCreation
        characterName="Test Character"
        characterClass="TestClass"
        characterGender="Female"
        style={styles}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
