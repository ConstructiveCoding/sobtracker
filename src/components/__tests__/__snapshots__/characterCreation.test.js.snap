// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCreation should render the character creation screen with no props 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Name
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize={true}
      autoCorrect={false}
      autoFocus={true}
      onChangeText={[Function]}
      placeholder="Name"
      testID="character-name-entry"
      value=""
    />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Class
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Class"
      testID="character-class-entry"
      value=""
    />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Gender
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Gender"
      testID="character-gender-entry"
      value=""
    />
  </View>
  <View
    testID="create-character-view"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Create Character
    </Text>
  </View>
</View>
`;

exports[`CharacterCreation should render the character creation screen with props 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Name
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize={true}
      autoCorrect={false}
      autoFocus={true}
      onChangeText={[Function]}
      placeholder="Name"
      testID="character-name-entry"
      value="Test Character"
    />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Class
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Class"
      testID="character-class-entry"
      value="TestClass"
    />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Gender
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Gender"
      testID="character-gender-entry"
      value="Female"
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="create-character-button"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Create Character
    </Text>
  </View>
</View>
`;
